plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts

```diff
@@ -114,6 +127,23 @@ export function createFetchTemplateAction(options: {
         );
       }
 
+      if (
+        ctx.input.extension &&
+        (ctx.input.copyWithoutRender || ctx.input.cookiecutterCompat)
+      ) {
+        throw new InputError(
+          'Fetch action input extension incompatible with copyWithoutRender and cookiecutterCompat',
+        );
+      }
+
+      let extension: string | false = false;
+      if (ctx.input.extension) {
+        extension =
+          typeof ctx.input.extension === 'boolean'
+            ? '.njk'
+            : ctx.input.extension;
```

Comment:
I agree with the problem. I'd rather not encourage ambiguous behavior, so I went in and just added some input validation. I also realized that I hadn't written tests for the other input checks, so I added those.

Reply:
+1 to @freben's suggestion to be forgiving here. Whether or not we're forgiving, perhaps we shouldn't include the `.` in the value we accept? Not sure whether this is universal, but I would generally expect to provide the string _after_ the period for a parameter using the term "extension", and the [definition on Wikipedia](https://en.wikipedia.org/wiki/Filename_extension) seems to support this.

Reply:
I made that choice based on hacking on editor modes a while back. https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide#contributing-a-basic-grammar includes the extension, https://flight-manual.atom.io/hacking-atom/sections/creating-a-grammar/#language-recognition actually says that the suffix might be the full file name, so it's really `endsWith` instead of just `extname` there.

Reply:
Fair enough! I guess just more confirmation that we should accept both :).

Reply:
Yeah, done that now. I pondered complaining in the logs, but didn't.